pkg/k8s/annotate.go

```diff
@@ -118,3 +121,20 @@ func (k8sCli K8sClient) AnnotateNode(nodeName string, encryptKey uint8, v4CIDR,
 
 	return nil
 }
+
+// GetSecrets returns the secrets found in the given namespace and name.
+func (k8sCli K8sClient) GetSecrets(ctx context.Context, ns, name string) (map[string][]byte, error) {
+	result := &coreV1.Secret{}
+	err := k8sCli.CoreV1().RESTClient().Get().
+		Context(ctx).
+		Namespace(ns).
+		Resource("secrets").
+		Name(name).
+		VersionedParams(&v1.GetOptions{}, scheme.ParameterCodec).
+		Do().
+		Into(result)
+	if err != nil {
+		return nil, err
+	}
+	return result.Data, nil
+}
```

Comment:
This also requires the following change:
```patch
diff --git a/install/kubernetes/cilium/charts/agent/templates/clusterrole.yaml b/install/kubernetes/cilium/charts/agent/templates/clusterrole.yaml
index d3b4c169f..a49ac1297 100644
--- a/install/kubernetes/cilium/charts/agent/templates/clusterrole.yaml
+++ b/install/kubernetes/cilium/charts/agent/templates/clusterrole.yaml
@@ -49,6 +49,12 @@ rules:
   - list
   - watch
   - update
+{{- if eq "k8s" .Values.global.tls.secretsBackend }}
+- apiGroups:
+  - ""
+  resources:
+  - secrets
+{{- end }}
 - apiGroups:
   - cilium.io
   resources:
diff --git a/install/kubernetes/cilium/values.yaml b/install/kubernetes/cilium/values.yaml
index 500ea8249..e17ac0f79 100644
--- a/install/kubernetes/cilium/values.yaml
+++ b/install/kubernetes/cilium/values.yaml
@@ -301,3 +301,6 @@ global:
 
   daemon:
     runPath: "/var/run/cilium"
+
+  tls:
+    secretsBackend: local
```

for which the users will be able to enable the RBAC for secrets access using

```
helm template cilium \
         --namespace=kube-system \
         --set global.tls.secretsBackend=k8s \
         ....
```

Reply:
This would negate the need for `CertificatesPath`, assuming we would map a secret `<namespace>/<name>` to a path in `CertsDirectory` when `secretsBackend: local`?

Reply:
I would actually favor that!

Reply:
Done.